enterprise/cmd/frontend/internal/registry/extension_manifest.go

```diff
@@ -32,35 +32,45 @@ func getExtensionManifestWithBundleURL(ctx context.Context, extensionID string,
 	if err != nil && !errcode.IsNotFound(err) {
 		return nil, time.Time{}, err
 	}
-	if release != nil {
-		// Add URL to bundle if necessary.
-		var o map[string]interface{}
-		if err := json.Unmarshal([]byte(release.Manifest), &o); err != nil {
-			return nil, time.Time{}, fmt.Errorf("parsing extension manifest for extension with ID %d (release tag %q): %s", registryExtensionID, releaseTag, err)
-		}
-		if o == nil {
-			o = map[string]interface{}{}
+	if release == nil {
+		return nil, time.Time{}, nil
+	}
+
+	if err := prepReleaseManifest(extensionID, release); err != nil {
+		return nil, time.Time{}, fmt.Errorf("parsing extension manifest for extension with ID %d (release tag %q): %s", registryExtensionID, releaseTag, err)
+	}
+
+	return &release.Manifest, release.CreatedAt, nil
+}
+
+// prepReleaseManifest will set the Manifest field of the release. If the manifest has no "url"
+// field itself, a "url" field pointing to the extension's bundle is inserted. It also returns
+// the date that the release was published.
+func prepReleaseManifest(extensionID string, release *dbRelease) error {
+	// Add URL to bundle if necessary.
+	var o map[string]interface{}
+	if err := json.Unmarshal([]byte(release.Manifest), &o); err != nil {
+		return err
+	}
+	if o == nil {
+		o = map[string]interface{}{}
+	}
+	urlStr, _ := o["url"].(string)
```

Comment:
the zero value here worries me. o is a map[string]interface{}. if "url" is not in o it returns nil. .(string) will make it panic

if you know for sure that it's a string if present then maybe something like this

```suggestion
	urlStr := ""
	if v, ok := o["url"]; ok {
	    urlStr = v.(string)
	}
```

Reply:
https://play.golang.org/p/1W6njcrOvSB :p

Reply:
It's fine to rely on the zero value: A two-position type assertion that fails promises to deliver a zero of the candidate type (https://golang.org/ref/spec#Type_assertions).

Reply:
got it, thanks :-)